src/Domain/Services/GoogleAnalytics.php

```diff
@@ -0,0 +1,99 @@
+<?php
+namespace Automattic\WooCommerce\Blocks\Domain\Services;
+
+use Automattic\WooCommerce\Blocks\Package;
+use Automattic\WooCommerce\Blocks\Assets\Api as AssetApi;
+
+/**
+ * Service class to integrate Blocks with the Google Analytics extension,
+ */
+class GoogleAnalytics {
+	/**
+	 * Constructor
+	 */
+	public function __construct() {
+		if ( ! Package::feature()->is_experimental_build() ) {
+			return;
+		}
+		add_action( 'init', array( $this, 'register_assets' ) );
+		add_action( 'wp_enqueue_scripts', array( $this, 'enqueue_scripts' ) );
+		add_filter( 'script_loader_tag', array( $this, 'async_script_loader_tags' ), 10, 3 );
+	}
+
+	/**
+	 * Register scripts.
+	 */
+	public function register_assets() {
+		$asset_api = Package::container()->get( AssetApi::class );
```

Comment:
I realize this is likely in place because of the thoughts around extracting this eventually into the GA plugin but until then, I think it'd be better to explicitly list `AssetApi` as a dependency in the constructor. While it'll have to be reworked anyways (either way) when moved to GA because the AssetApi class only works for paths in the blocks plugin, I think by moving to the constructor now, we're promoting the kind of patterns we want to see in code living in the repo?

This applies for `enqueue_scripts` as well.